include/ts/ts.h

```diff
@@ -2353,6 +2353,16 @@ tsapi const char *TSVConnSslProtocolGet(TSVConn sslp);
 */
 tsapi const char *TSVConnSslCurveGet(TSVConn sslp);
 
+/**
+   Return the current (if set) SSL Server Name. This is still owned by the
+   core, and must not be free'd.
+
+   @param txnp the transaction pointer
+
+   @return the SSL Curve
+*/
+tsapi const char *TSVConnSslServerNameGet(TSVConn sslp);
```

Comment:
1. update docs
2. update tests. Would be perfect to cover for different protocols, including H1, H2 and Quic.